### ISSUE TITLE
Evaluate the run time of various simulations

### ISSUE THREAD

**Issue by astrojarred** [astrojarred/grbsens#4]
Once parallelization is set up (issue #1), evaluate the runtime of simulations of various observation durations. 

- use the `timeit` python package

**Comment by astrojarred** [comment]
Results from running `timeit` on simulations with durations of varying orders of magnitude (OOM):

```python
t_int [s]    avg runtime of 9 runs
‾‾‾‾‾‾‾‾‾   ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
   10e0        2min  9s ±  36ms
   10e1       12min 12s ± 818ms
   10e2        1min 12s ±  61ms
   10e3        1min 35s ±  31ms
   10e4        2min 27s ± 227ms
```

Runtime does not vary too much across OOMs. However, there is a peculiarity at `t_int=10s`.
Running simulations around this value shows that this is a unique situation to this single run:

```python
t_int [s]    runtime [s]
‾‾‾‾‾‾‾‾‾   ‾‾‾‾‾‾‾‾‾‾‾‾‾
    6            74
    7            74
    8            92
    9           110
   10           733 
   11            91 
   12           109 
   13           109 
   14            92
```

In further examining the log, there seems to be an issue with the fitting algorithm.  It eventually recovers, but just takes some extra time:

```python
For t_int=10s:
iteration n.    simulated sens [mCrab]
‾‾‾‾‾‾‾‾‾‾‾‾    ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
      1                       100.0000
      2                       135.2406
      3                 200961523.9819
      4                       310.4297
      5                       264.8123
      6                       299.5544
      7                       281.7662
  converged                   271.1229
```

So in the end I don't think it is affecting our results too much, however I think after running the first full simulation it will be worth checking how often this issue persists.

All this in mind, It seems that the average simulation time lies somewhere in the range of 1.5-2 mins regardless of the observation time.
If we assume we want to do 1000 iterations on duration, each taking an average of 2 minutes, that would take:
33 h on 1 core
33 min on 60 cores (leaving 4 cores free)

In a very pessimistic situation, perhaps 10% of the simulations will have a similar issue to that of the 10s, and take 15 min instead of 2 min. In this case:
55 h on 1 core
55 min on 60 cores

Once parallelization is implemented (issue #1), we can run a full simulation and further evaluate runtimes.

**Comment by astrojarred** [comment]
With the `timeit` package, I ran a simulation involving 1120 observations ranging from 0.25 - 10000s:

With three iterations of five runs each, the average run time was 00:31:01 ± 00:00:0079

For approximation purposes, it's reasonable to say that each simulation will take around half an hour, with the main factor being the observation time resolution required. That said, other input parameters might have some effect on the total run time as well.
